File: hw/quiroga/U2/Hw18Exceptions/Inheritance_JS/javascript/ec/edu/espe/zoo/model/Food.mjs
export class Food {
    #id;
    #amount;
    #type;

    constructor(id, amount, type) {
        this.#id = id;
        this.#amount = amount;
        this.#type = type;
    }

    toString() {
        return `Food {id: ${this.#id}, amount: ${this.#amount}, type: ${this.#type}}`;
    }

    getId() {
        return this.#id;
    }

    setId(id) {
        this.#id = id;
    }

    getAmount() {
        return this.#amount;
    }

    setAmount(amount) {
        this.#amount = amount;
    }

    getType() {
        return this.#type;
    }

    setType(type) {
        this.#type = type;
    }
}

File: ws/proano/u2/Ws15Inheritance/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Food.java


package ec.edu.espe.zooapp.model;

/**
 *
 * @author Alan Arias
 */
public class Food {
    private int id;
    private float amount;
    private String type;

    public Food(int id, float amount, String type) {
        this.id = id;
        this.amount = amount;
        this.type = type;
    }

    @Override
    public String toString() {
        return "Food{" + "id=" + id + ", amount=" + amount + ", type=" + type + '}';
    }
    
    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the amount
     */
    public float getAmount() {
        return amount;
    }

    /**
     * @param amount the amount to set
     */
    public void setAmount(float amount) {
        this.amount = amount;
    }

    /**
     * @return the type
     */
    public String getType() {
        return type;
    }

    /**
     * @param type the type to set
     */
    public void setType(String type) {
        this.type = type;
    }
    
}
